### ISSUE TITLE
JavaScript error in Tutor 18.0, 18.1, 18.1.1, 18.1.2, 18.1.3 - users unable to access account page. 

### ISSUE THREAD

**Issue by pkolyvas** [overhangio/tutor#1111]
**Bug description**

There is an issue with Tutor v18.0+ (18.0, 18.1.0, 18.1.1, 18.1.2 and 18.1.3) where the accounts page (MFE app) fails to load as a result of some issues with JavaScript assets. 

Please see this screenshot. This occurs when someone tries to view their account in order to validate or edit their user profile:
![Screenshot 2024-08-13 at 8 53 21 AM](https://github.com/user-attachments/assets/ddbddef7-81f0-4b39-8779-fea959e514ed)
![Screenshot 2024-08-13 at 9 50 06 AM](https://github.com/user-attachments/assets/c0087dcb-c864-46ca-932e-02e910fbe2ef)

The logs provide no indication of error, in part because, as best as we can tell, the root cause is on the client side assets.  The browser console (tested on Chrome, Firefox, Safari) provides the same error:
```
Error: Minified React error #130; visit https://reactjs.org/docs/error-decoder.html?invariant=130&args[]=undefined&args[]= for the full message or use the non-minified dev environment for full errors and additional helpful warnings.
    React 16
    80492 index.jsx:27
    o pubsub.js:67
    i pubsub.js:88
    a pubsub.js:99
    setTimeout handler*u pubsub.js:146
    publish pubsub.js:160
    s pubSub.js:46
    e initialize.js:361
    f initialize.js:2
    _ initialize.js:2
    k initialize.js:2
    Vt initialize.js:2
    i initialize.js:2
    promise callback*Vt initialize.js:2
    i initialize.js:2
    promise callback*Vt initialize.js:2
    i initialize.js:2
    promise callback*Vt initialize.js:2
    i initialize.js:2
    promise callback*Vt initialize.js:2
    i initialize.js:2
    promise callback*Vt initialize.js:2
    i initialize.js:2
    promise callback*Vt initialize.js:2
    i initialize.js:2
    promise callback*Vt initialize.js:2
    i initialize.js:2
    promise callback*Vt initialize.js:2
    i initialize.js:2
    promise callback*Vt initialize.js:2
    i initialize.js:2
    Ht initialize.js:2
    Ht initialize.js:2
    on initialize.js:369
    rn initialize.js:288
    80492 index.jsx:58
    Webpack 6
react-dom.production.min.js:216:199
```

**How to reproduce**

Reproduction is possible on a vanilla, completely unchanged Tutor install. This is true for both someone installing via PIP as well as customers using the AWS AMI Marketplace image. Please see this thread for even more context: https://discuss.openedx.org/t/problem-displaying-account-content-after-upgraded-to-redwood-with-tutor-v18/13641

I use a script to install tutor, as part of spinning up the EC2 instance via Terraform. The userdata script is here and can be easily converted into a shell script for reproduction: https://gist.github.com/pkolyvas/331ac52029801cd38190600ffecb05f2

**Environment**

- AWS Marketplace AMI upgrade
- Ubuntu 24.04LTS/amd64/Tutor 18.0,18.1,18.1.1,18.1.2 & 18.1.3 confirmed.

**Comment by regisb** [comment]
1. What plugins are you running? `tutor plugins list`
2. What is your OS and browser version?

**Comment by pkolyvas** [comment]
1. None other than stock as per the linked reproduction script:
```
PLUGINS:
- forum
- indigo
- mfe
```
2. MacOS / Unbuntu 22.04 / Windows 11. All browsers. Edge, Chrome, Firefox. 
3. Don't have a running 18.x install at the moment. I've asked others in the thread linked above to include their information here as well.

**Comment by regisb** [comment]
Please paste the output of `tutor plugins list`. This command is giving some useful information, including the plugin versions.

**Comment by Tyler-SEM** [comment]
I've got the same issue on a fresh local tutor install.

1. tutor plugins list
android         installed       18.0.0
cairn           installed       18.0.0
credentials     installed       18.0.0
discovery       installed       18.0.0
ecommerce       installed       18.0.0
forum           installed       18.0.0
indigo          ✅ enabled      18.1.0
jupyter         installed       18.0.0
mfe             ✅ enabled      18.0.0
minio           installed       18.0.0
notes           installed       18.0.0
webui           installed       18.0.0
xqueue          installed       18.0.0

2. Windows 11, Firefox 129.0.1 / Chome 127.0.6533.100 / Edge 127.0.2651.98 

3. tutor mounts list
[]

**Comment by regisb** [comment]
Am I correct in assuming that you are *not* facing that issue on https://apps.sandbox.openedx.edly.io/account/ ?
Login: admin / admin or student / student

**Comment by Tyler-SEM** [comment]
Correct, that link works fine.

**Comment by pkolyvas** [comment]
Alright I rolled _another_ fresh installation of 18.1.3 (using the provided script above). Same issue. 

AWS EC2, Canonical's 24.04LTS AMI image on an M7i-flex.large instance, with a public IP (no reverse proxy / load-balancer in front of the instance).

1. `tutor plugins list`:
```
NAME            	STATUS   	VERSION
android         	installed	18.0.0
cairn           	installed	18.0.0
credentials     	installed	18.0.0
discovery       	installed	18.0.0
ecommerce       	installed	18.0.0
forum           	✅ enabled	18.0.0
indigo          	✅ enabled	18.1.0
jupyter         	installed	18.0.0
mfe             	✅ enabled	18.0.0
minio           	installed	18.0.0
notes           	installed	18.0.0
third-party-auth	installed	1.0.0
webui           	installed	18.0.0
xqueue          	installed	18.0.0
```

2. MacOS Sonoma 14.6, Firefox 129 (but also Windows 11 24H2, FF 129)

3. `tutor mounts list`:
`[]`

I have a complete set of logs available from the installation all the way to the error here: https://gist.githubusercontent.com/pkolyvas/aa8d0967d62b850abc8595e57645b699/raw/5a056c1e559f13408d94c5dbb91766611c9b1691/tutor.log

**Comment by regisb** [comment]
Yes, I finally managed to reproduce the issue. But I haven't figured out why the issue was not happening on my local setup, and why it's not happening on sandbox.openedx.edly.io either. This is almost certainly an issue related to Docker caching during image building. @hinakhadim is still investigating...

**Comment by jennyq** [comment]
> * What plugins are you running? `tutor plugins list`
> 
> * What is your OS and browser version?
> 
> * What is the output of `tutor mounts list`?

1) `tutor plugins list`

  ```
  NAME       	STATUS   	VERSION
    android    	installed	18.0.0
    cairn      	installed	18.0.0
    credentials	installed	18.0.0
    discovery  	installed	18.0.0
    ecommerce  	installed	18.0.0
    forum      	✅ enabled	18.0.0
    indigo     	✅ enabled	18.1.0
    jupyter    	installed	18.0.0
    mfe        	✅ enabled	18.0.0
    minio      	installed	18.0.0
    notes      	installed	18.0.0
    webui      	installed	18.0.0
    xqueue     	installed	18.0.0
```

2) Ubuntu 24.04 LTS

3) `tutor mounts list`
[]

**Comment by pkolyvas** [comment]
> Yes, I finally managed to reproduce the issue. But I haven't figured out why the issue was not happening on my local setup, and why it's not happening on sandbox.openedx.edly.io either. This is almost certainly an issue related to Docker caching during image building. @hinakhadim is still investigating...

Thanks for all your help. :)

**Comment by regisb** [comment]
> I solved the problem by just install the tutor by clone the git and used pip install -e . instead of "pip install tutor[full]"

Nope, this is not going to resolve anything. Don't waste your time.

**Comment by acaleechurn** [comment]
Is there a time frame to get this issue fixed?

**Comment by hinakhadim** [comment]
@acaleechurn PR has been merged to resolve this issue. You can get the fix in `tutor-indigo=18.1.1`.

**Comment by DawoudSheraz** [comment]
https://github.com/overhangio/tutor-indigo/pull/96 has resolved this.
